### ISSUE TITLE
Create a Text Editor 

### ISSUE THREAD

**Issue by aritra-tech** [aritra-tech/Notify#38]
Create a Text Editor in AddNoteScreen where there will be buttons for Bold, Italic, Strikethrough, etc.

**Comment by githubashutoshsoni** [comment]
Can you please assign this to me?

**Comment by aritra-tech** [comment]
Sure @githubashutoshsoni 
All the best.

**Comment by githubashutoshsoni** [comment]
@aritra-tech we have handful of libraries for compose yet they are not very customizing. can you suggest some library that does this or we need to build our own custom compose library for rich edit text? I checked out the below it does most of the job but  BaseEditText does not have onValueChange function that can return the result 

[](https://github.com/MohamedRejeb/Compose-Rich-Editor)

**Comment by aritra-tech** [comment]
I suggest not to use a third party library.

**Comment by githubashutoshsoni** [comment]
Okay, I've got it working. Can you please merge single tap to top to develop branch since I got this completed as well. I will update this once I you do a quick merge to develop branch so there are no conflicts.

**Comment by aritra-tech** [comment]
I have requested some changes in the PR that you have created please check it out. Once it's fixed I will merge it out.

**Comment by aritra-tech** [comment]
Hey @githubashutoshsoni any updates?

**Comment by githubashutoshsoni** [comment]
I haven't resolved the issue yet. The creation of a new note functions correctly, but editing an existing note seems to be malfunctioning. Could you please review it in this pull request?  #141 

      RichTextStyleRow(
                    modifier = Modifier.fillMaxWidth(),
                    state = basicRichTextState,
                )

                RichTextEditor(
                    state = basicRichTextState,
                    modifier = Modifier
                        .fillMaxSize(),
                    textStyle = TextStyle(
                        fontSize = 18.sp,
                        fontFamily = FontFamily(Font(R.font.poppins_light)),
                    ),
                    keyboardOptions = KeyboardOptions.Default.copy(
                        capitalization = KeyboardCapitalization.Sentences,
                        keyboardType = KeyboardType.Text,
                    ),

                    maxLines = Int.MAX_VALUE,
                    placeholder = {
                        Text(
                            stringResource(R.string.notes),
                            fontSize = 20.sp,
                            fontWeight = FontWeight.W500,
                            color = Color.Gray,
                            fontFamily = FontFamily(Font(R.font.poppins_light))
                        )
                    },
                )
                description = basicRichTextState.annotatedString.text
                if (isNew) {
                    characterCount = title.length + description.length
                } else {
                    addEditViewModel.updateDescription(description)
                }

**Comment by aritra-tech** [comment]
Hey @githubashutoshsoni can you please fix the changes that I have told you regarding the PR that you have raised.

**Comment by Ismail-AD** [comment]
@aritra-tech i am back assign it to me and also have you uploaded any post about changes in v1.5.1 on LinkedIn

**Comment by aritra-tech** [comment]
Yeah sure 
All the best 👋

**Comment by Ismail-AD** [comment]
I have tried it today the thing is it will change all the text style rather the upcoming typed text and the library of RICH text compose will change the total structure of your description text field function which i don't want to do !!!!

**Comment by aritra-tech** [comment]
Which library did you use?
This one - https://github.com/MohamedRejeb/Compose-Rich-Editor

**Comment by Ismail-AD** [comment]
yes this one
But the library of RICH text compose will change the total structure of your description text field function which i don't want to do !!!!

**Comment by aritra-tech** [comment]
Okay got your point.

**Comment by andy-ife** [comment]
Hi @aritra-tech. Want me to look into this?

**Comment by aritra-tech** [comment]
yeah sure @andy-ife

**Comment by andy-ife** [comment]
What features would you like added? Here are the ones I think are most important:
- bold
- italic
- underline
- hyperlink
- strikethrough
- justify
- color text
Also, whatever property edited with the text editor will be stored with the note in db right?

**Comment by aritra-tech** [comment]
yeah this features are good for now and yes it will be stored in the db

**Comment by andy-ife** [comment]
Alright. You've already said you'd prefer not to use a third party library. I'll research into how it can be done using vanilla compose. Concerning persistence, what do you think of adding another table to the db? The text properties could get much (isBold, isItalic, isHyperlink, isStrikethrough etc), so I'd add them to a new table called 'note properties' with a foreign key pointing to original note table. What do you think?
